src/Backups/BackupInfo.cpp

```diff
@@ -98,4 +99,23 @@ String BackupInfo::toStringForLogging() const
     return toAST()->formatForLogging();
 }
 
+void BackupInfo::copyS3CredentialsTo(BackupInfo & dest) const
+{
+    /// named_collection case, no need to update
+    if (!dest.id_arg.empty() || !id_arg.empty())
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "use_same_s3_credentials_for_base_backup is not compatible with named_collections");
+
+    auto & dest_args = dest.args;
+
+    if ((dest_args.size() != 1) && (dest_args.size() != 3))
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Backup S3 requires 1 or 3 arguments: url, [access_key_id, secret_access_key]");
+
+    // 0 - s3_uri
+    if (dest_args.size() >= 3)
+    {
+        dest_args[1] = args[1];
+        dest_args[2] = args[2];
```

Comment:
And if `dest_args.size()==1` then we should append two arguments to `dest_args`.